lib/Interpreter/IncrementalParser.cpp

```diff
@@ -125,7 +125,7 @@ namespace {
   if (!II)
     return;
   const clang::DefMacroDirective* MD
-    = llvm::dyn_cast<clang::DefMacroDirective>(PP.getMacroDirective(II));
+    = llvm::dyn_cast_or_null<clang::DefMacroDirective>(PP.getMacroDirective(II));
```

Comment:
cling -x c
Really the code is already set up to handle CLING_CXXABIS not being defined (see EarlyReturnWarn struct in the same function)